src/System.Net.Primitives/src/System/Net/NetworkInformation/NetworkInformationException.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.ComponentModel;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+namespace System.Net.NetworkInformation
+{
+    /// <devdoc>
+    ///    <para>
+    ///       Provides NetworkInformation exceptions to the application.
+    ///    </para>
+    /// </devdoc>
+    public class NetworkInformationException : Win32Exception
```

Comment:
That’s what I understood as well. I completely agree on the Win32Exception name being unfortunate… Unless we decide to change (which I was told we will not for app-compat reasons), there’s nothing we could really do here.

From: Stephen Toub [mailto:notifications@github.com]
Sent: Friday, July 31, 2015 5:28 AM
To: dotnet/corefx <corefx@noreply.github.com>
Cc: Cristian Pop <Cristian.Pop@microsoft.com>
Subject: Re: [corefx] Initial commit of System.Net.Primitives. (#2559)


In src/System.Net.Primitives/src/System/Net/NetworkInformation/NetworkInformationException.cs<https://github.com/dotnet/corefx/pull/2559#discussion_r35968691>:

> +// Copyright (c) Microsoft. All rights reserved.

> +// Licensed under the MIT license. See LICENSE file in the project root for full license information.

> +

> +using System;

> +using System.ComponentModel;

> +using System.Net.Sockets;

> +using System.Runtime.InteropServices;

> +

> +namespace System.Net.NetworkInformation

> +{

> +    /// <devdoc>

> +    ///    <para>

> +    ///       Provides NetworkInformation exceptions to the application.

> +    ///    </para>

> +    /// </devdoc>

> +    public class NetworkInformationException : Win32Exception

@CIPop<https://github.com/CIPop>, sorry, I wasn't being very clear: I'd made a few comments about Win32Exception in previous networking-related PRs on GitHub, not in this one. The comment was simply asking whether Win32Exception is the right base class. We've made Win32Exception work on Unix, essentially wrapping an errno value and doing the appropriate strerror string translations, but it's an unfortunately named type.

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/corefx/pull/2559/files#r35968691>.